src/register_target.cpp

```diff
@@ -32,6 +32,28 @@
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
+#ifdef _WIN32
+namespace {
+struct auto_load_targets
+{
+    auto_load_targets()
+    {
+        make_target("ref");
+#ifdef HAVE_CPU
+        make_target("cpu");
+#endif
+#ifdef HAVE_GPU
+        make_target("gpu");
+#endif
+#ifdef HAVE_FPGA
+        make_target("fpga");
+#endif
+    }
+};
+[[maybe_unused]] static auto load_targets = auto_load_targets{};
+} // namespace
```

Comment:
This needs to go into a static/object library that we only link in for the tests. This prevents lazy loading the targets.